vapoursynth/src/api.rs

```diff
@@ -675,10 +788,26 @@ impl API {
     /// # Safety
     /// The caller must ensure `core` is valid.
     #[inline]
+    #[cfg(not(feature = "gte-vapoursynth-api-36"))]
     pub(crate) unsafe fn get_core_info(self, core: *mut ffi::VSCore) -> *const ffi::VSCoreInfo {
         (self.handle.as_ref().getCoreInfo)(core)
     }
 
+    /// Returns information about the VapourSynth core.
+   ///
+   /// # Safety
+   /// The caller must ensure `core` is valid.
+    #[inline]
+    #[cfg(feature = "gte-vapoursynth-api-36")]
+    pub(crate) unsafe fn get_core_info(self, core: *mut ffi::VSCore) -> *const ffi::VSCoreInfo {
+        use std::mem::MaybeUninit;
+
+        let mut core_info = MaybeUninit::<ffi::VSCoreInfo>::uninit();
+        (self.handle.as_ref().getCoreInfo2)(core, core_info.as_mut_ptr());
+        core_info.assume_init();
+        core_info.as_ptr()
```

Comment:
You're returning a pointer to a struct sitting on the stack of this function; its lifetime ends as soon as the function returns.

I think it's better to do this by returning the `ffi::VSCoreInfo` itself from this version, and then adding a `cfg` over here:
https://github.com/YaLTeR/vapoursynth-rs/blob/a2e46f584520bb249400b2ce0867750a1175c471/vapoursynth/src/core.rs#L68-L73
with something along this:
```rust
        #[cfg(not(feature = "gte-vapoursynth-api-36"))]
        let raw_info = unsafe {
            API::get_cached()
                .get_core_info(self.handle.as_ptr())
                .as_ref()
                .unwrap()
        };
        #[cfg(feature = "gte-vapoursynth-api-36")]
        let raw_info = unsafe {
            &API::get_cached()
                .get_core_info(self.handle.as_ptr())
        };
```